sktime/forecasting/statsmodels_arima.py

```diff
@@ -0,0 +1,316 @@
+        every specification (for example 'yule_walker' can only be used with
+        AR(p) models).
+
+    method_kwargs : dict, optional Arguments to pass to the fit function for the
+        parameter estimator described by the method argument.
+
+    gls : bool, optional
+        Whether to use generalized least squares (GLS) to estimate regression effects.
+        The default is False if method='statespace' and is True otherwise.
+
+    gls_kwargs : dict, optional Arguments to pass to the GLS estimation fit method.
+        Only applicable if GLS estimation is used (see gls argument for details).
+
+    cov_type : str, optional The cov_type keyword governs the method for calculating
+        the covariance matrix of parameter estimates. Can be one of 'opg' for the outer
+        product of gradient estimator, 'oim' for the observed information matrix
+        estimator, calculated using the method of Harvey (1989), 'approx' for the
+        observed information matrix estimator, calculated using a numerical
+        approximation of the Hessian matrix, 'robust' for an approximate
+        (quasi-maximum likelihood) covariance matrix that may be valid even in the
+         presence of some misspecifications. Intermediate calculations use the 'oim'
+         method. 'robust_approx' is the same as 'robust' except that the intermediate
+         calculations use the 'approx' method. 'none' for no covariance matrix
+         calculation. Default is 'opg' unless memory conservation is used to avoid
+         computing the loglikelihood values for each observation,
+         in which case the default is 'oim'.
+
+    cov_kwds : dict or None, optional
+        A dictionary of arguments affecting covariance matrix computation.
+
+    return_params : bool, optional
+        Whether to return only the array of maximizing parameters. Default is False.
+
+    low_memory : bool, optional If set to True, techniques are applied to
+        substantially reduce memory usage. If used, some features of the results object
+        will not be available (including smoothed results and in-sample prediction),
+        although out-of-sample forecasting is possible. Default is False.
+
+    See Also
+    --------
+    ARIMA
+    AutoARIMA
+    StatsForecastAutoARIMA
+
+    Examples
+    --------
+    >>> from sktime.datasets import load_airline
+    >>> from sktime.forecasting.statsmodels_arima import StatsModelsARIMA
+    >>> y = load_airline()
+    >>> forecaster = StatsModelsARIMA(order=(0, 0, 12))  # doctest: +SKIP
+    >>> forecaster.fit(y)  # doctest: +SKIP
+    >>> y_pred = forecaster.predict(fh=[1,2,3])  # doctest: +SKIP
+    """
+
+    _tags = {
+        # packaging info
+        # --------------
+        "authors": ["arnaujc91"],
+        "maintainers": ["arnaujc91"],
+        "ignores-exogeneous-X": False,
+        "capability:pred_int": True,
+        "capability:pred_int:insample": True,
+        "python_dependencies": ["statsmodels"],
```

Comment:
This is a very nitpicky comment, but we probably need some lower bound here. I created a `python:3.8-slim-bullseye` container, which is the lowest `sktime` supported `python` version, and installed `"statsmodels==0.12.1`, which the lowest `sktime` supported `statsmodels` version. The following failed:

```pycon
root@b4a11a0add98:/# python
Python 3.8.18 (default, Jan 11 2024, 11:45:27)
[GCC 10.2.1 20210110] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> from statsmodels.tsa.arima.model import ARIMA
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/usr/local/lib/python3.8/site-packages/statsmodels/tsa/arima/model.py", line 12, in <module>
    from statsmodels.tsa.statespace import sarimax
  File "/usr/local/lib/python3.8/site-packages/statsmodels/tsa/statespace/sarimax.py", line 19, in <module>
    from statsmodels.tsa.arima.specification import SARIMAXSpecification
  File "/usr/local/lib/python3.8/site-packages/statsmodels/tsa/arima/specification.py", line 11, in <module>
    from statsmodels.tsa.base.tsa_model import TimeSeriesModel
  File "/usr/local/lib/python3.8/site-packages/statsmodels/tsa/base/tsa_model.py", line 7, in <module>
    from pandas import (to_datetime, Int64Index, DatetimeIndex, Period,
ImportError: cannot import name 'Int64Index' from 'pandas' (/usr/local/lib/python3.8/site-packages/pandas/__init__.py)
>>>
```

Can you please crosscheck, and if you also face this, may be it's possible to find a specific lower bound where this starts working. I think it's removed in `pandas` v2 ([reference](https://pandas.pydata.org/pandas-docs/version/1.5/reference/api/pandas.Int64Index.html)).

Or may be @fkiraly can recommend if we can ignore such specific version combination issues altogether.

Reply:
@yarnabrina, I was trying to find the minimal version combinations here, for a "lower bound" test element:
https://github.com/sktime/sktime/pull/5148

If you have a container ready, would be appreciated if you could post the lowest functioning versions you found.

Reply:
I am confused how to define minimum for a pair (statsmodels version, pandas version), sometimes even numpy version comes into play. I have observed these works on python 3.8 container:

statsmodels==0.12.1, pandas<2, numpy<1.24 (with deprecation warnings)
statsmodels==0.13.2, pandas==2.0.3

(whereever I've said "<", haven't checked explitly there)

Reply:
Should we take this?

```
statsmodels==0.13.2, pandas==2.0.3
```

Reply:
> I am confused how to define minimum for a pair (statsmodels version, pandas version)

My opinion: only a combination that is mutually compatible, and the highest such combination.

Say you define `statsmodels <= X.Y.Z` and `numpy <= A.B.C`, and that ends up installing `statsmodels <= D.E.F` and `numpy <= A.B.C`, then it is the latter combination you take.

If there are only inequality constraints at any point in time, and these are monotonous, there is a pareto-maximal such combination.

Reply:
imo there is only one combination to try if you have a guess at a minimum for both:

1. install env with both you guessed minimums
2. read out the versions that get installed (resulting from dep resolution)

Reply:
I also don't think I'll have time explore, but if the above (try two and choose) does not work, my simplistic suggestion will the support from current version onwards, so 0.14+ and 2.2+ where the current CI is actively testing it.

Reply:
Trying to figure out how to get the minimum compatible version combo with `statsmodels` and `pandas`. But i do not see how to do it without checking each combination. 

If i do `pip install 'statsmodels>0.12.1'` will automatically install the latest possible version of statsmodels.